src/csm_testing/tests/sat_functional/hwinv/test_hwinv.py

```diff
@@ -0,0 +1,389 @@
+        command = 'sat hwinv --list-cabinet-pdu-power-connectors'
+        self.run_hwinv_test(command, 'cabinet pdu power connectors')
+
+    def test_hwinv_list_all(self) -> None:
+        """Test that `sat hwinv --list-all --show-missing --show-empty` validates only the components present in the system."""
+        # Get the list of components present in the system
+        present_components = get_present_components()
+
+        command = 'sat hwinv --list-all --show-missing --show-empty'
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        stdout_lines = stdout.splitlines()
+
+        # Iterate through the lines to find sections and validate headers
+        for i, line in enumerate(stdout_lines):
+            # Match section titles like "Listing of all <SOME COMPONENT TYPE>"
+            match = re.match(r"Listing of all (.+)", line.strip())
+            if not match:
+                continue
+
+            section_name = match.group(1)
+            lower_sec   = section_name.lower()
+            # Skip any section type not reported as present
+            if lower_sec not in present_components:
+                continue
+
+            self.assertIn(section_name, HWINV_HEADERS,
+                f"Unexpected section '{section_name}' found in output."
+            )
+
+            # Extract the expected header
+            expected_header = HWINV_HEADERS[section_name]
+
+            # Get the header line (3 lines below the section title)
+            header_line_index = i + 3
+            if header_line_index >= len(stdout_lines):
+                self.fail(f"Header row for section '{section_name}' not found in output.")
+            header_line = stdout_lines[header_line_index].strip()
+
+            # Check for empty tables (no header or data rows)
+            if not header_line.startswith("|"):
+                continue
+
+            # Parse the actual header
+            actual_header = [col.strip() for col in header_line.split("|") if col.strip()]
+
+            # Validate the actual header against the adjusted expected header
+            self.assertEqual( expected_header, actual_header,
+                f"Header for section '{section_name}' does not match the expected fields. "
+                f"Expected: {expected_header}, Actual: {actual_header}"
+            )
+
+    def test_hwinv_list_node_fields(self) -> None:
+        """Test that `sat hwinv --list-nodes --node-fields 'xname, FRUID, Serial Number, SKU'` returns the proper header."""
+        command = "sat hwinv --list-nodes --node-fields 'xname, FRUID, Serial Number, SKU'"
+        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
+        stdout = proc.stdout.decode()
+        expected_header = ['xname', 'FRUID', 'Serial Number', 'SKU']
```

Comment:
Minor suggestion to make it unnecessary to specify the fields twice:

```suggestion
        expected_header = ['xname', 'FRUID', 'Serial Number', 'SKU']
        command = f"sat hwinv --list-nodes --node-fields '{",".join(expected_header)}'"
        proc = subprocess.run(shlex.split(command), stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=True)
        stdout = proc.stdout.decode()
```